common/Geometry2d/RectTest.cpp

```diff
@@ -18,16 +20,63 @@ TEST(Rect, hit) {
     EXPECT_FALSE(example.hit(Point(1, -0.091)));
 
     // degenerate case
-    Rect example2(Point(0, 0), Point(0, 0));
-    EXPECT_TRUE(example2.hit(Point(0, 0)));
-    EXPECT_TRUE(example2.hit(Point(0, 0.09)));
-    EXPECT_FALSE(example2.hit(Point(0, 0.091)));
-    EXPECT_FALSE(example2.hit(Point(1, 1)));
+    EXPECT_TRUE(nullExample.hit(Point(0, 0)));
+    EXPECT_TRUE(nullExample.hit(Point(0, 0.09)));
+    EXPECT_FALSE(nullExample.hit(Point(0, 0.091)));
+    EXPECT_FALSE(nullExample.hit(Point(1, 1)));
 }
 
 TEST(Rect, containsPoint) {
     EXPECT_TRUE(example.containsPoint(Point(0, 0.5)));
     EXPECT_FALSE(example.containsPoint(Point(-1, 01)));
 }
 
-}  // namespace Geometry2d
+/* Tests below can be uncommented to test if this function breaks
+ * Needs to be made public for tests to work.
+TEST(Rect, cohenCodes){
+    EXPECT_EQ(intersectExample.CohenSutherlandOutCode(Point(0,1.2)), 0x00);
+    EXPECT_EQ(intersectExample.CohenSutherlandOutCode(Point(-1,1)), 0x00);
+    EXPECT_EQ(intersectExample.CohenSutherlandOutCode(Point(-1.5,1.5)), 0x01);
+    EXPECT_EQ(intersectExample.CohenSutherlandOutCode(Point(1.5,1.2)), 0x02);
+    EXPECT_EQ(intersectExample.CohenSutherlandOutCode(Point(1.5,2)), 0x02);
+    EXPECT_EQ(intersectExample.CohenSutherlandOutCode(Point(-.5,0)), 0x04);
+    EXPECT_EQ(intersectExample.CohenSutherlandOutCode(Point(-1,0)), 0x04);
+    EXPECT_EQ(intersectExample.CohenSutherlandOutCode(Point(-1,3)), 0x08);
+    EXPECT_EQ(intersectExample.CohenSutherlandOutCode(Point(-.5,3)), 0x08);
+    EXPECT_EQ(intersectExample.CohenSutherlandOutCode(Point(-1.1,0)), 0x05);
+    EXPECT_EQ(intersectExample.CohenSutherlandOutCode(Point(1.1,0)), 0x06);
+    EXPECT_EQ(intersectExample.CohenSutherlandOutCode(Point(-1.5,2.5)), 0x09);
+    EXPECT_EQ(intersectExample.CohenSutherlandOutCode(Point(1.5,2.5)), 0x0A);
+}*/
+
+TEST(Rect, degenerageCohenCodes){
+    EXPECT_EQ(nullExample.CohenSutherlandOutCode(Point(0,0)), 0x00);
+    EXPECT_EQ(nullExample.CohenSutherlandOutCode(Point(1,0)), 0x02);
+    EXPECT_EQ(nullExample.CohenSutherlandOutCode(Point(-1,0)), 0x01);
+    EXPECT_EQ(nullExample.CohenSutherlandOutCode(Point(0,1)), 0x08);
+    EXPECT_EQ(nullExample.CohenSutherlandOutCode(Point(0,-.1)), 0x04);
+}
```

Comment:
This function needs to be commented out too since `CohenSutherlandOutCode` is private

Reply:
Done